orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RestartStageHandler.kt

```diff
@@ -58,13 +59,38 @@ class RestartStageHandler(
         if (topStage.status.isComplete) {
           topStage.addRestartDetails(message.user)
           topStage.reset()
+          restartParentPipelineIfNeeded(message, topStage)
           repository.updateStatus(topStage.execution.type, topStage.execution.id, RUNNING)
           queue.push(StartStage(startMessage))
         }
       }
     }
   }
 
+  private fun restartParentPipelineIfNeeded(message: RestartStage, topStage: Stage) {
+    val trigger = topStage.execution.trigger
+    if (!topStage.execution.trigger.type.equals("pipeline")) {
```

Comment:
you are probably better of checking the type of the trigger in a type-safe way
```suggestion
    if (!topStage.execution.trigger instanceof PipelineTrigger) {
```